src/imagery/i.eodag/i.eodag.py

```diff
@@ -799,6 +829,47 @@ def products_compare(first, second):
     return search_result
 
 
+def skip_existing(output, search_result):
+    """Remove products that is already downloaded and saved in 'output' directory.
+
+    :param output: Output directory whose files will be compared with the scenes.
+    :type output: class'eodag.api.search_result.SearchResult'
+
+    :param search_results: EO products to be checked for existence in 'output' directory.
+    :type search_result: class'eodag.api.search_result.SearchResult'
+
+    :return: Sorted EO products
+    :rtype: class:'eodag.api.search_result.SearchResult'
+    """
+    filtered_result = []
+    # Check for previously downloaded scenes
+    existing_files = [os.path.join(output, f) for f in os.listdir(output)]
+    if len(existing_files) <= 1:
+        return search_result
+    # TODO: Implement a way to check if a file is completely downloaded
+    #       or if it was interrupted
+    for scene in search_result:
+        existing_file = [
+            sfile for sfile in existing_files if scene.properties["title"] in sfile
+        ]
+        if existing_file:
+            creation_time = datetime.utcfromtimestamp(
+                os.path.getctime(existing_file[0])
+            )
+            ingestion_time = datetime.fromisoformat(
+                scene.properties["startTimeFromAscendingNode"]
+            ).replace(tzinfo=None)
+            if ingestion_time <= creation_time:
+                gs.message(
+                    _("Skipping scene: {} which is already downloaded.").format(
+                        scene.properties["title"]
+                    )
```

Comment:
Why check for creation and ingestion time to decide to skip a file or not, isn't the matching file name enough? I do not understand the logic of comparing creation and ingestion time of existing files.

Reply:
I wasn't sure why we need to compare creation and ingestion time as well, but I wrote in that way to maintain the logic of the previous implementation in i.sentinel.download.
i.sentinel.download had the same comparison done when skipping scenes, though I can't think of any scenario where that comparison would matter... Shall I remove it?

Reply:
> When data gets re-processed (e.g. with a new processing baseline) then users may have downloaded a product with the previous processing baseline, but the provider has a newer version of the same product. That has been the motivation behind the original check.
> 
> That said, I do not think _"startTimeFromAscendingNode"_ translates to what has been _"ingestiondate"_. Here properties like _"modificationDate"_ or _"publicationDate"_ are more appropriate I guess. Not sure if all providers support those...

Ah I see that makes sense now, thanks for the explanation. 

> When data was downloaded with _sentinelsat_ "in-progess" files got a suffix and then renamed when downloaded completely. So, `(Path(output) / scene.properties["title"]).exists()` would not find incomplete downloads... How does eodag handle filenames during download?

When eodag downloads a file it just uses the scene's ID (aka. title) without any suffixes. So there isn't a direct way of checking if the product was completely downloaded or not. But there is another method... we could use the hidden directory `.download` which is automatically created by eodag when it downloads a scene.
Inside this directory, eodag creates a file for each product **_that is completely downloaded_**, and saves the source URL of that product. So we could use that to detect if a product was completely downloaded or if it was interrupted.

Reply:
> we could use the hidden directory `.download` which is automatically created by eodag when it downloads a scene. Inside this directory, eodag creates a file for each product **_that is completely downloaded_**, and saves the source URL of that product. So we could use that to detect if a product was completely downloaded or if it was interrupted.

Nice! I did not know. Maybe worth to handle incomplete downloads in an "atexit" procedure?

Reply:
> Maybe worth to handle incomplete downloads in an "atexit" procedure?

Sounds good, will give it a try.

Reply:
@ninsbl I implemented the check for whether to skip a scene or not based on the existence of the corresponding file in `.downloaded`, and I have pushed these changes.

For the "atexit" procedure, there is a scenario that I think can be quiet dangerous:
What if a user fully download a scene Q in directory X, and then they moved that scene manually to directory Y.
Later on, if they decide to run i.eodag with the `output` option pointing to directory Y, and the same scene Q manages to appear again in the search result... now if scene Q fails to download this time.. Both the fully downloaded scene and the partially downloaded will be deleted...

I think multiple similar scenarios might accidentally delete unwanted file?